console/console-init/ui/src/modules/project-detail/components/AccessCredentials/AccessCredentials.tsx

```diff
@@ -82,164 +75,177 @@ const AccessCredentials: React.FunctionComponent<IAccessCredentialsProps> = ({
-    <>
-      <Text component={TextVariants.h2}>Messaging</Text>
-      <br />
-      {messaging.url && (
-        <InputText
-          label={"Tenant Id"}
-          type={"text"}
-          value={messaging.url}
-          isReadOnly={true}
-          enableCopy={true}
-          ariaLabel={"input-messaging-url"}
-        />
-      )}
-      <br />
-      {messaging.username && (
-        <InputText
-          label={"User name"}
-          type={"text"}
-          value={messaging.username}
-          isReadOnly={true}
-          enableCopy={true}
-          ariaLabel={"input-messaging-username"}
-        />
-      )}
-      <br />
-      {messaging.password && (
-        <InputText
-          label={"Password"}
-          type={"password"}
-          value={messaging.password}
-          isReadOnly={true}
-          ariaLabel={"input-messaging-password"}
-        />
-      )}
-      <br />
-      {messaging.addressSpace && (
-        <InputText
-          label={"Address Space"}
-          type={"text"}
-          value={messaging.addressSpace}
-          isReadOnly={true}
-          enableCopy={true}
-          ariaLabel={"input-messaging-addressSpace"}
-        />
-      )}
-      <br />
-      {messaging.eventsAddressName && (
-        <InputText
-          label={"Events address name"}
-          type={"text"}
-          value={getCombinedString(
-            messaging.eventsAddressName.type,
-            messaging.eventsAddressName.plan
-          )}
-          isReadOnly={true}
-          enableCopy={true}
-          isExpandable={messaging.eventsAddressName.plan ? true : false}
-          ariaLabel={"input-messaging-eventsAddressName"}
-        />
-      )}
-      <br />
-      {messaging.telemetryAddressName && (
-        <InputText
-          label={"Telemetry address name"}
-          type={"text"}
-          value={getCombinedString(
-            messaging.telemetryAddressName.type,
-            messaging.telemetryAddressName.plan
-          )}
-          isReadOnly={true}
-          enableCopy={true}
-          isExpandable={messaging.telemetryAddressName.plan ? true : false}
-          ariaLabel={"input-messaging-telemetryAddressName"}
-        />
-      )}
-      <br />
-      {messaging.commandAddressName && (
-        <InputText
-          label={"Command address name"}
-          type={"text"}
-          value={getCombinedString(
-            messaging.commandAddressName.type,
-            messaging.commandAddressName.plan
-          )}
-          isReadOnly={true}
-          enableCopy={true}
-          isExpandable={messaging.commandAddressName.plan ? true : false}
-          ariaLabel={"input-messaging-commandAddressName"}
-        />
-      )}
-      <br />
-    </>
-  );
+
+  const Messaging = () => {
+    const {
+      url,
+      username,
+      password,
+      addressSpace,
+      commandAddressName,
+      eventsAddressName,
+      telemetryAddressName
+    } = messaging && messaging;
```

Comment:
why && condition for same level object i.e. messaging && messaging;
if you facing compile time or time issue, undefined properties, safer side you can follow below syntax.
   const {
      url,
      username,
      password,
      addressSpace,
      commandAddressName,
      eventsAddressName,
      telemetryAddressName
    } = messaging || {};
In this case, if object is undefined it will not give undefined property issue.